torch/library.py

```diff
@@ -0,0 +1,87 @@
+from ._ops import OpOverload
+from typing import Set
+import traceback
+import torch._C as C
+__all__ = ['extend_library']
+
+# User created libraries to extend existing libraries
+# Each user created library is added here to ensure that it's not automatically removed outside the
+# scope of the function it was created in.
+_impls_for_existing_libraries = {}
+
+# Set containing the combination of (namespace, operator, DispatchKey) for which a new kernel has been registered
+# The keys in the set are of the form `namespace + "/" + op_name + "/" + dispatch_key`.
+# This set is maintained to ensure that two libraries don't try to override the exact same functionality to avoid
+# libraries calling into kernels not intended to be called.
+_impls: Set[str] = set()
+
+class _Library:
```

Comment:
Not sure why this class is private tbh. We return it and expect the user to manipulate it.